jetcd-core/src/main/java/io/etcd/jetcd/WatchImpl.java

```diff
@@ -195,7 +195,9 @@ public void onNext(WatchResponse response) {
         String reason = response.getCancelReason();
         Throwable error;
 
-        if (Strings.isNullOrEmpty(reason)) {
+        if (response.getCompactRevision() != 0) {
```

Comment:
This could also be done like this:
Leave old cold and add the following:

`} else if (response.getCanceled() && response.getCompactRevision() == 0) {`